site/docs/encoding.md

```diff
@@ -34,127 +50,177 @@ An integral part of the data visualization process is encoding data with visual
+In addition, field definitions for different encoding channels may support the following properties:
 
-{% include table.html props="detail" source="Encoding" %}
+- [`scale`](scale.html) - The function that transforms values in the data domain (numbers, dates, strings, etc) to visual values (pixels, colors, sizes) for [position](#position) and [mark property](#mark-prop) channels.
 
-**Note**: Since `detail` represents an actual data field in the aggregation, it cannot encode a constant `value`.
+- [`axis`](axis.html) - The guiding visualization to aid interpretation of scales for [position channels](#position).
 
-#### Examples
+- [`legend`](legend.html) - The guiding visualization to aid interpretation of [mark property channels](#mark-prop).
 
-Here is a scatterplot showing average horsepower and displacement for cars from different origins. We map `Origin` to `detail` channel to use the field as a group-by field without mapping it to visual properties of the marks.
+- [`format`](format.html) - The formatting pattern for text value for [text channels](#text).
 
-<div class="vl-example" data-name="scatter_aggregate_detail"></div>
+- [`stack`](stack.html) - Type of stacking offset if a [position field](#position) with continuous domain should be stacked.
 
+- [`sort`](sort.html) - Sort order for a field for [position](#position) and [mark property](#mark-prop) channels.
 
-Here is a line chart showing stock prices of 5 tech companies over time.
-We map `symbol` variable (stock market ticker symbol) to `detail` to use them to group lines.
+- [`condition`](condition.html) - The conditional encoding rule for [mark property](#mark-prop) and [text](#text) channels.
 
-<div class="vl-example" data-name="line_detail"></div>
+To see a list of additional properties for each type of encoding channels, please see the specific sections
+for [position](#position), [mark property](#mark-prop), [text](#text), [detail](#detail), [order](#order), and [facet](#facet) channels.
 
-<!-- TODO Need to decide if we want to keep the two examples above since they look bad with labels / tooltips -->
+{:#value-def}
+### Value Definition
 
+{: .suppress-error}
+```json
+// Specification of a Single View
+{
+  ...,
+  "encoding": {     // Encoding
+    ...: {
+      "value": ...
+    },
+    ...
+  },
+  ...
+}
+```
 
-### Mark Order Channels
+To map a constant value to an encoding channel, the channel's value definition must describe the `value` property.
+(See the [`value`](value.html) page for more examples.)
 
-`order` channel sorts the layer order or stacking order (for stacked charts) of the marks while `path` channel sorts the order of data points in line marks.
+{% include table.html props="value" source="NumberValueDef" %}
 
-{% include table.html props="order" source="Encoding" %}
+TODO: replace type with `Any` (as the type of `value` depends on channel.)
```

Comment:
@domoritz  Do we have a way to replace Number as Any just for `NumberValueDef`?

Alternatively, we can make this single table manually.